crates/engine/tree/src/tree/precompile_cache.rs

```diff
@@ -2,9 +2,12 @@
 
 use reth_evm::precompiles::{DynPrecompile, Precompile};
 use revm::precompile::{PrecompileOutput, PrecompileResult};
-use revm_primitives::Bytes;
+use revm_primitives::{Address, Bytes, HashMap};
 use std::sync::Arc;
 
+/// Stores caches for each precompile.
+pub type PrecompileCacheMap = HashMap<Address, PrecompileCache>;
```

Comment:
please no aliases for this, this makes it incredibly hard to trace where we access this type

this should just be its own type